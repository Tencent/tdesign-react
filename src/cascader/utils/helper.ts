import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
