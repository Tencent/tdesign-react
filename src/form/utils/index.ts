import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
