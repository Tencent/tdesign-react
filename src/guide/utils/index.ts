import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
