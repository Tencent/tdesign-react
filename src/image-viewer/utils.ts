export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
