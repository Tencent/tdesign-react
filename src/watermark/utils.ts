const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
