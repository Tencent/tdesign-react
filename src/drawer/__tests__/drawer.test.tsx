import React, { useState } from 'react';
import { fireEvent, render, testExamples } from '@test/utils';
import Drawer from '../index';

// 测试组件代码 Example 快照
testExamples(__dirname);

function DrawerDemo(props) {
  const [visible, setVisible] = useState(false);

  const handleClick = () => {
    setVisible(true);
  };
  const handleClose = () => {
    setVisible(false);
  };
  return (
    <>
      <div onClick={handleClick}>Open</div>
      <Drawer
        header={props.header || '这是标题'}
        footer={props.footer}
        visible={visible}
        onClose={handleClose}
        confirmBtn={props.confirmBtn}
        cancelBtn={props.cancelBtn}
        showOverlay={props.notShowOverlay ? false : true}
        mode={props.mode || 'overlay'}
        placement={props.placement || 'right'}
        attach={props.attach}
        destroyOnClose={props.destroyOnClose || false}
        size={props.size || 'small'}
        onCancel={props.onCancel}
      >
        <p>This is a drawer</p>
      </Drawer>
    </>
  );
}

describe('test Drawer', () => {
  test('Drawer render', () => {
    const { container, getByText } = render(<DrawerDemo />);
    expect(container.querySelector('.t-drawer--open')).not.toBeInTheDocument();
    fireEvent.click(getByText('Open'));
    expect(container.querySelector('.t-drawer--open')).toBeInTheDocument();
    expect(container.querySelector('.t-drawer__mask')).toBeInTheDocument();
    expect(container.querySelector('.t-drawer__content-wrapper--right')).toBeInTheDocument();
    expect(container.querySelector('.t-drawer__content-wrapper--top')).not.toBeInTheDocument();
  });
  test('Drawer close', () => {
    const { container, getByText } = render(<DrawerDemo />);
    fireEvent.click(getByText('Open'));
    fireEvent.click(container.querySelector('.t-drawer__close-btn'));
    expect(container.querySelector('.t-drawer--open')).not.toBeInTheDocument();
    expect(container.querySelector('.t-drawer')).toBeInTheDocument();
  });
  test('Drawer mask close', () => {
    const { container, getByText } = render(<DrawerDemo />);
    fireEvent.click(getByText('Open'));
    expect(container.querySelector('.t-drawer--open')).toBeInTheDocument();
    fireEvent.click(container.querySelector('.t-drawer--open').children[0]);
    expect(container.querySelector('.t-drawer--open')).not.toBeInTheDocument();
  });
  test('Drawer placement', () => {
    const { container, getByText } = render(<DrawerDemo placement="top" />);
    fireEvent.click(getByText('Open'));
    expect(container.querySelector('.t-drawer__content-wrapper--right')).not.toBeInTheDocument();
    expect(container.querySelector('.t-drawer__content-wrapper--top')).toBeInTheDocument();
  });
  test('Drawer size', () => {
    const { container, getByText } = render(<DrawerDemo size="50%" />);
    fireEvent.click(getByText('Open'));
    expect(container.querySelector('.t-drawer__content-wrapper--right')).toHaveStyle({ width: '50%' });
  });
  test('Drawer showOverlay', () => {
    const { container, getByText } = render(<DrawerDemo notShowOverlay />);
    fireEvent.click(getByText('Open'));
    expect(container.querySelector('.t-drawer__mask')).not.toBeInTheDocument();
  });
  test('Drawer header and footer', () => {
    const { getByText } = render(<DrawerDemo header={<div>自定义头部</div>} footer={<div>自定义底部</div>} />);
    fireEvent.click(getByText('Open'));
    expect(getByText('自定义头部').parentElement).toHaveClass('t-drawer__header');
    expect(getByText('自定义底部').parentElement).toHaveClass('t-drawer__footer');
  });
  test('Drawer cancelBtn and confirmBtn', () => {
    const { getByText } = render(<DrawerDemo cancelBtn={<div>cancelBtn</div>} confirmBtn={<div>confirmBtn</div>} />);
    fireEvent.click(getByText('Open'));
    expect(getByText('cancelBtn').parentElement.parentElement).toHaveClass('t-drawer__footer');
    expect(getByText('confirmBtn').parentElement.parentElement).toHaveClass('t-drawer__footer');
  });
  test('Drawer mode push', () => {
    const { getByText } = render(<DrawerDemo attach="body" mode="push" />);
    fireEvent.click(getByText('Open'));
    expect(document.body).toHaveStyle({
      margin: '0 0 0 -300px',
    });
  });
  test('Drawer onCancel', () => {
    const onCancelFn = jest.fn();
    const { getByText } = render(<DrawerDemo onCancel={onCancelFn} />);
    fireEvent.click(getByText('Open'));
    expect(onCancelFn).not.toBeCalled();
    fireEvent.click(getByText('取消'));
    expect(onCancelFn).toBeCalled();
  });
});
