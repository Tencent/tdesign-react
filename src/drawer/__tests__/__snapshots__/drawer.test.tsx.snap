// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attach-parent.jsx 1`] = `
<DocumentFragment>
  <div
    style="position: relative; height: 400px; padding: 48px; overflow: hidden; background: rgb(250, 250, 250); border: 1px solid #ebedf0; border-radius: 2px;"
  >
    <div
      class="tdesign-demo-block-column"
      style="transition: margin 300ms cubic-bezier(0.7, 0.3, 0.1, 1) 0s;"
    >
      <p>
        渲染在当前元素中。
      </p>
      <div>
        <span>
          抽屉弹出方向：
        </span>
        <div
          class="t-radio-group t-size-m t-radio-group__outline"
        >
          <label
            class="t-radio-button"
          >
            <input
              class="t-radio-button__former"
              type="radio"
              value="left"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              左侧
            </span>
          </label>
          <label
            class="t-radio-button t-is-checked"
          >
            <input
              checked=""
              class="t-radio-button__former"
              type="radio"
              value="right"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              右侧
            </span>
          </label>
          <label
            class="t-radio-button"
          >
            <input
              class="t-radio-button__former"
              type="radio"
              value="top"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              上方
            </span>
          </label>
          <label
            class="t-radio-button"
          >
            <input
              class="t-radio-button__former"
              type="radio"
              value="bottom"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              下方
            </span>
          </label>
        </div>
      </div>
      <div>
        <span>
          抽屉弹出模式：
        </span>
        <div
          class="t-radio-group t-size-m t-radio-group__outline"
        >
          <label
            class="t-radio-button t-is-checked"
          >
            <input
              checked=""
              class="t-radio-button__former"
              type="radio"
              value="push"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              push
            </span>
          </label>
          <label
            class="t-radio-button"
          >
            <input
              class="t-radio-button__former"
              type="radio"
              value="overlay"
            />
            <span
              class="t-radio-button__input"
            />
            <span
              class="t-radio-button__label"
            >
              overlay
            </span>
          </label>
        </div>
      </div>
      <div>
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Open
          </span>
        </button>
      </div>
      <div>
        <div
          class="t-drawer t-drawer--right t-drawer--attach"
          tabindex="-1"
        >
          <div
            class="t-drawer__mask"
          />
          <div
            class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
            style="width: 300px;"
          >
            <div
              class="t-drawer__close-btn"
            >
              <svg
                class="t-icon t-icon-close"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <div
              class="t-drawer__header"
            >
              Drawer
            </div>
            <div
              class="t-drawer__body"
            >
              <p>
                This is a drawer
              </p>
            </div>
            <div
              class="t-drawer__footer"
            >
              <div
                style="display: flex; justify-content: flex-start;"
              >
                <button
                  class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                  type="button"
                >
                  <span
                    class="t-button__text"
                  >
                    确认
                  </span>
                </button>
                <button
                  class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                  type="button"
                >
                  <span
                    class="t-button__text"
                  >
                    取消
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        打开抽屉
      </span>
    </button>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          >
            抽屉标题
          </div>
          <div
            class="t-drawer__body"
          >
            <p>
              抽屉的内容
            </p>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`custom.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        打开抽屉
      </span>
    </button>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          >
            <div>
              自定义头部
            </div>
          </div>
          <div
            class="t-drawer__body"
          >
            <div>
              自定义内容
            </div>
          </div>
          <div
            class="t-drawer__footer"
          >
            <button
              class="t-button t-button--theme-primary t-button--variant-base"
              type="button"
            >
              <span
                class="t-button__text"
              >
                自定义底部按钮
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`destroy.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        打开抽屉
      </span>
    </button>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          />
          <div
            class="t-drawer__body"
          >
            <p>
              抽屉的内容
            </p>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`no-mask.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        打开抽屉
      </span>
    </button>
    <div>
      <div
        class="t-drawer t-drawer--right t-drawer--without-mask"
        tabindex="-1"
      >
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          >
            抽屉标题
          </div>
          <div
            class="t-drawer__body"
          >
            <p>
              抽屉的内容
            </p>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`operation.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        打开抽屉
      </span>
    </button>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          >
            抽屉标题
          </div>
          <div
            class="t-drawer__body"
          >
            <form
              class="t-form"
            >
              <div
                class="t-form__item"
              >
                <div
                  class="t-form__label t-form__label--right"
                  style="width: 60px;"
                >
                  <label>
                    Label A
                  </label>
                </div>
                <div
                  class="t-form__controls"
                  style="margin-left: 60px;"
                >
                  <div
                    class="t-form__controls-content"
                  >
                    <div
                      class="t-input__wrap"
                    >
                      <div
                        class="t-input t-size-m t-align-left t-is-default"
                      >
                        <input
                          class="t-input__inner"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="t-form__item"
              >
                <div
                  class="t-form__label t-form__label--right"
                  style="width: 60px;"
                >
                  <label>
                    Label B
                  </label>
                </div>
                <div
                  class="t-form__controls"
                  style="margin-left: 60px;"
                >
                  <div
                    class="t-form__controls-content"
                  >
                    <div
                      class="t-input__wrap"
                    >
                      <div
                        class="t-input t-size-m t-align-left t-is-default"
                      >
                        <input
                          class="t-input__inner"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`placement.jsx 1`] = `
<DocumentFragment>
  <div>
    <div
      class="t-radio-group t-size-m t-radio-group__outline"
    >
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="left"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          左侧
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          checked=""
          class="t-radio-button__former"
          type="radio"
          value="right"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          右侧
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="top"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          上方
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="bottom"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          下方
        </span>
      </label>
    </div>
    <div>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin-top: 20px;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          打开抽屉
        </span>
      </button>
    </div>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          />
          <div
            class="t-drawer__body"
          >
            <p>
              抽屉的内容
            </p>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popup.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div>
      <span>
        抽屉弹出模式：
      </span>
      <div
        class="t-radio-group t-size-m t-radio-group__outline"
      >
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            checked=""
            class="t-radio-button__former"
            type="radio"
            value="push"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            push
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            type="radio"
            value="overlay"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            overlay
          </span>
        </label>
      </div>
    </div>
    <div>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        type="button"
      >
        <span
          class="t-button__text"
        >
          打开抽屉
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`size.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-radio-group t-size-m t-radio-group__outline"
    >
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          checked=""
          class="t-radio-button__former"
          type="radio"
          value="small"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          small(300px)
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="medium"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          medium(500px)
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="large"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          large(760px)
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="200"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          200
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="400px"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          400px
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          type="radio"
          value="50%"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          50%
        </span>
      </label>
    </div>
    <div>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        type="button"
      >
        <span
          class="t-button__text"
        >
          打开抽屉
        </span>
      </button>
    </div>
    <div>
      <div
        class="t-drawer t-drawer--right"
        tabindex="-1"
      >
        <div
          class="t-drawer__mask"
        />
        <div
          class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
          style="width: 300px;"
        >
          <div
            class="t-drawer__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-drawer__header"
          />
          <div
            class="t-drawer__body"
          >
            <p>
              抽屉的内容
            </p>
          </div>
          <div
            class="t-drawer__footer"
          >
            <div
              style="display: flex; justify-content: flex-start;"
            >
              <button
                class="t-drawer__confirm t-button t-button--theme-primary t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  确认
                </span>
              </button>
              <button
                class="t-drawer__cancel t-button t-button--theme-default t-button--variant-base"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  取消
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
